### ISSUE TITLE
Assign dashboards to specific folder

### ISSUE THREAD

**Issue by till** [cloudalchemy/ansible-grafana#174]
Any thoughts on this? I see folder is hardcoded, currently: 
https://github.com/cloudalchemy/ansible-grafana/blob/aef4b7ec306fec9872bfabca47807c9cc2f83ecc/tasks/dashboards.yml#L119

Maybe we can add a "folder"?

My trouble is that whatever dashboards I install via this role, I can't "save" them in Grafana. E.g. moving them to a different folder, etc.. Which I think is a good thing.

**Comment by paulfantom** [comment]
Right now minimum supported ansible version for this role is 2.5. That ansible version includes python modules to handle provisioning of grafana dashboards and I am currently considering ditching two separate ways of managing dashboards with this role and use ansible modules. This means ditching provisioning dashboards as files by using grafana_use_provisioning as in current state this role grew too complex to be easily maintainable.

**Comment by paulfantom** [comment]
So, good idea, but I will remove this code pretty soon.

**Comment by till** [comment]
Hmm... I am pretty confused by:
https://docs.ansible.com/ansible/latest/modules/grafana_dashboard_module.html
https://docs.ansible.com/ansible/latest/plugins/lookup/grafana_dashboard.html

Which one would you use? Not entirely sure what a module and what a plugin is, and why the same name.

I guess you're saying that unless the module supports it, we can't use it?

**Comment by paulfantom** [comment]
`lookup` just reads data from somewhere, in this case from grafana instance. `module` manages the entity.

It would be good to use https://docs.ansible.com/ansible/latest/modules/grafana_dashboard_module.html. If it lacks something, then we can develop it in ansible.

I don't want to remove the option to provision dashboards as files, but this will need to be done out of scope of the role. Probably by some other role or on a playbook level.

Right now we got into a place where this role uses a lot of hacks just to manage dashboards in grafana instances. We are using `command` module to download dashboards from grafana.net, we have TWO ways of inserting said dashboards into grafana, and we even have some contraception which allows synchronizing files between deployer and target without `synchronize` module. All this is great when it can be supported and maintained. Sadly, as of now, the role is maintained only by me as a side project and I don't have much time to dwell into all edge cases provided by grafana itself. This is mostly a reason why all those contraceptions need to go away and the role should focus on providing an opinionated installation path.

**Comment by till** [comment]
@paulfantom I understand the idea to maintain less. But developing more inside an Ansible module slows down everything for me. Then the constraint is Ansible releases?

**Comment by paulfantom** [comment]
> I understand the idea to maintain less

That's correct. Additionally, idea is to move maintenance of frequently used parts into ansible itself as more people are using ansible modules to manage grafana than there are users of this role. I am assuming this will increase quality of this role.

> But developing more inside an Ansible module slows down everything for me.

Sorry about that. As a maintainer, I prioritize quality over velocity and I believe that's a good thing if we want to have a stable role in the long term.

> Then the constraint is Ansible releases?

As of now ansible releases aren't constraining anything. They were, a couple months ago, when 2.8 wasn't released. Right now main constrain is my lack of time :slightly_smiling_face:

**Comment by till** [comment]
Ok, let me rephrase — any improvement to the `ansible_grafana` module won't be immediately available? Or I need to run Ansible `master`?

**Comment by paulfantom** [comment]
That is correct. Improvement to ansible modules will be available when running form ansible `master` branch. Those won't be immediately available.

**Comment by till** [comment]
There's lots of open issues and PRs with `grafana_dashboard`. Do you know how maintained it is? I can't tell.

**Comment by paulfantom** [comment]
There are 7 open issues (3 of those by contributors to this role :tada:). 3 issues already have PRs in place and seuf is pretty good at answering to issues. So I would say that it is pretty well maintained.

**Comment by till** [comment]
@paulfantom I made the first PR to the Ansible module.

Following up with one to download Grafana dashboards (haven't published it yet). Is there anything else you want to get rid off here? Then Ansible version would be `2.10` at least — assuming my PR lands.

**Comment by paulfantom** [comment]
> Is there anything else you want to get rid off here?

As of today - no. :)

**Comment by till** [comment]
@paulfantom do you know how to get feedback on the PR?

**Comment by till** [comment]
Figured it - mostly asking people in `#ansible-devel` on Freenode.

Support for folders will be in `2.10`, can we use this? Or does this role require have minimum requirements for a different Ansible version?
